src/bin/fish_indent.rs

```diff
@@ -571,7 +573,7 @@ impl<'source, 'ast> PrettyPrinterState<'source, 'ast> {
         }
     }
 
-    fn emit_node_text(&mut self, node: &dyn Node) {
+    fn emit_node_text(&mut self, node: impl Node) {
```

Comment:
I think it's surprising that this is `impl Node` and not `&impl Node`. It works because this is a `NodeRef` in the concrete case.
Maybe it's better to be globally consistent about this (unless there are cases where we don't pass NodeRef?).

Reply:
This is a manifestation of what I consider a wart in my design, one that also appears in the form of the inherent `accept` method.
`&'a IfStatement` and `BranchRef<'a>` are both one indirection. `&NodeRef<'_>` is a double indirection. The type system doesn't really "understand" this, leaving us in our current situation.

I suspect things will get a little better as I make further changes to the design, but for now, I have no strong feelings about how to approach this.

Commit `5d9659ce464df78477fb0e7287f0e2106e293546` rids the codebase of any text matching the regex `: impl (ast::)?(Node|Keyword|Token)`, *mostly* by inserting `&` but in some cases I replaced the `impl Trait` with a specific "by-value ref-flavored enum".
Is this satisfactory?